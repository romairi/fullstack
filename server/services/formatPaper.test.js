const {formatPaper} = require('./formatPaper');

const idMock = 'MOCK_ID';
const linksMock = [{href: 'http://arxiv.org/abs/2005.12184v2', rel: 'alternate', type: 'text/html'},
    {title: 'pdf', href: 'http://arxiv.org/pdf/2005.12184v2', rel: 'related', type: 'application/pdf'}];


describe('formatPaper', () => {
    it('should return formatted paper without any link', () => {
        const res = formatPaper({
            id: idMock,
            links: [{href: 'http://arxiv.org/abs/2005.12184v2', rel: 'alternate', type: 'text/html'}],
            someArg: '123'
        });
        expect(res).toMatchSnapshot();
    });

    it('should return formatted paper without any link - the links arg is no an array', () => {
        const res = formatPaper({
            id: idMock,
            links: 'MOCK_LINK',
            someArg: '123'
        });
        expect(res).toMatchSnapshot();
    });

    it('should return formatted paper including link', () => {
        const res = formatPaper({
            id: idMock,
            links: linksMock,
            someArg: '123'
        });
        expect(res).toMatchSnapshot();
    });
    it('should return formatted paper without additional args', () => {
        const res = formatPaper({
            id: idMock,
            links: linksMock,
        });
        expect(res).toMatchSnapshot();
    });
});

