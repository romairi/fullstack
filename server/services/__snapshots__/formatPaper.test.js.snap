// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`formatPaper should return formatted paper including link 1`] = `
Object {
  "paperId": "MOCK_ID",
  "pdfLink": "http://arxiv.org/pdf/2005.12184v2",
  "someArg": "123",
}
`;

exports[`formatPaper should return formatted paper without additional args 1`] = `
Object {
  "paperId": "MOCK_ID",
  "pdfLink": "http://arxiv.org/pdf/2005.12184v2",
}
`;

exports[`formatPaper should return formatted paper without any link - the links arg is no an array 1`] = `
Object {
  "paperId": "MOCK_ID",
  "pdfLink": null,
  "someArg": "123",
}
`;

exports[`formatPaper should return formatted paper without any link 1`] = `
Object {
  "paperId": "MOCK_ID",
  "pdfLink": null,
  "someArg": "123",
}
`;
